unified_addrs.py

```diff
@@ -0,0 +1,173 @@
+            s = 2
+        elif s == 2:
+            if len(acc) < receiver_len:
+                acc.append(b)
+            
+            if len(acc) == receiver_len:
+                if receiver_type == 0 or receiver_type == 1:
+                    assert not ('transparent' in result), "duplicate transparent receiver detected"
+                    assert len(acc) == 20
+                    result['transparent'] = bytes(acc)
+                    acc = []
+                    s = 0
+
+                elif receiver_type == 2:
+                    assert not ('sapling' in result), "duplicate sapling receiver detected"
+                    assert len(acc) == 43
+                    result['sapling'] = bytes(acc)
+                    acc = []
+                    s = 0
+
+                elif receiver_type == 3:
+                    assert not ('orchard' in result), "duplicate orchard receiver detected"
+                    assert len(acc) == 43
+                    result['orchard'] = bytes(acc)
+                    acc = []
+                    s = 0
+    return result
+
+
+def main():
+    args = render_args()
+
+    from random import Random
+    rng = Random(0xabad533d)
+    def randbytes(l):
+        ret = []
+        while len(ret) < l:
+            ret.append(rng.randrange(0, 256))
+        return bytes(ret)
+    rand = Rand(randbytes)
+
+    test_vectors = []
+    for _ in range(0, 10):
+        has_t_addr = rand.bool()
+        if has_t_addr:
+            t_addr = b"".join([rand.b(20)])
+        else:
+            t_addr = None
+
+        has_s_addr = rand.bool()
+        if has_s_addr:
+            sapling_sk = sapling_key_components.SpendingKey(rand.b(32))
+            sapling_default_d = sapling_sk.default_d()
+            sapling_default_pk_d = sapling_sk.default_pkd()
+            sapling_raw_addr = b"".join([sapling_default_d[:11], bytes(sapling_default_pk_d)[:32]])
+        else:
+            sapling_raw_addr = None
+
+        has_o_addr = (not has_s_addr) or rand.bool()
+        if has_o_addr:
+            orchard_sk = orchard_key_components.SpendingKey(rand.b(32))
+            orchard_fvk = orchard_key_components.FullViewingKey(orchard_sk)
+            orchard_default_d = orchard_fvk.default_d()
+            orchard_default_pk_d = orchard_fvk.default_pkd()
+            orchard_raw_addr = b"".join([orchard_default_d[:11], bytes(orchard_default_pk_d)[:32]])
+        else: 
+            orchard_raw_addr = None
+
+        receivers = [
+            orchard_raw_addr, 
+            sapling_raw_addr, 
+            (rand.bool(), t_addr)
+        ]
+        ua = encode_unified(receivers)
+
+        decoded = decode_unified(ua)
+        assert decoded.get('orchard') == orchard_raw_addr
+        assert decoded.get('sapling') == sapling_raw_addr
+        assert decoded.get('transparent') == t_addr
+
+        test_vectors.append({
+            't_addr_bytes': t_addr,
```

Comment:
To use these test vectors, we need to know what kind of t-addr is expected:
```suggestion
            'p2pkh_bytes': t_addr if is_p2pkh else None,
            'p2sh_bytes': None if is_p2pkh else t_addr,
```